src/Parser.js

```diff
@@ -45,7 +45,13 @@ import ParseError from "./ParseError";
 
 /**
  * An initial function (without its arguments), or an argument to a function.
- * The `result` argument should be a ParseNode.
+ *
+ * @param {ParseNode|string} result It's a string if `isFunction=true` or if
+ *     `result="$"` for switching into math mode; otherwise, it's a ParseNode
+ *     and `isFunction` must be false.
+ * @param {boolean} isFunction True when this is a function. False when it's a
+ *     function argument or "$" to switch into math mode.
+ * @param {Token} token
  */
 function ParseFuncOrArgument(result, isFunction, token) {
```

Comment:
This should really be specified as a class since that's how it's being used.  The fact that `result` can sometimes be a `string` and sometimes a `ParseNode` makes things hard to reason about.  No need to do anything about it in this PR though.

Reply:
I completely agree about the point of confusion. Once we have a static type system in place, I'm rather inclined to do a little refactor. But I'd open an issue on that first to open up discussion.